### ISSUE TITLE
Add contract id to the position

### ISSUE THREAD

**Issue by holzeis** [get10101/10101#1921]
At the moment we are assuming the user only ever has one "active" position in the database. e.g. proposed, rollover, closing, etc. But due to bugs it can happen that the user ends up with multiple positions in proposed, etc. 

In other words our approach on identifying the position related to the dlc channel is very failure prone.

In order to prevent that I propose we add the contract id to the position. That way we can easily find the position based on a dlc message / dlc channel and know with certainty that we are working on the correct position.

**Comment by holzeis** [comment]
E.g. We have 3 open positions by a single user (@bonomat). I presume we ended up in this state as he successfully rejected his subchannel offer and managed to open a new positions. Since we did not update the failed positions to `Failed`, all positions remained `Proposed` and have been set to `Open` once the last order was successfully processed.

**Comment by bonomat** [comment]
Proposal: we tackle this in two iterations: 
1. we add the contract id, e.g. during rollover
2. we redesign our approach and add the contract id during creation

**Comment by bonomat** [comment]
@holzeis : please copy in the diagram

**Comment by holzeis** [comment]
Adding the contract id is unfortunately not as easy as we would like it to be for several reasons. For one there is no unique id that can always be trusted to identify a dlc channel or contract.

- **Problem**: No real connection between a dlc channel / contract and a position.
- Connection is only based on an assumption.

- The position is a logical extension of a contract
- The dlc channel is merely a vessel to move / transaction on a contract / position.

In the discussions we concluded that the trade is far under developed, with the position model having taking over far too many responsibilities. That lead to the original reason for opening this ticket.

- The positions should not have a state. Its are view on the history of trades or contract transactions. And should always be reconstructable from that data.

----

Below diagram illustrates some thought processes on how a 10101 can better reflect the rust-dlc states. Goal is to be able to imply all 10101 states from the underlying rust-dlc data models.

**Challenges**
- Have a unique id linking 10101 and rust-dlc data.

![Image](https://github.com/get10101/10101/assets/382048/3ee051c6-ef04-4b63-b16d-60a7b8dc7e33)

![Image](https://github.com/get10101/10101/assets/382048/3ec780a9-9555-48e9-a296-358a5265acf2)

----

Closing this ticket in favor of smaller more concrete tickets.
